__tests__/map-set.js

```diff
@@ -0,0 +1,176 @@
+"use strict"
+import {Immer, nothing, original, isDraft, immerable} from "../src/index"
+import {each, shallowCopy, isEnumerable, DRAFT_STATE} from "../src/common"
+
+jest.setTimeout(1000)
+
+runBaseTest("proxy (no freeze)", true, false)
+runBaseTest("proxy (autofreeze)", true, true)
+runBaseTest("proxy (autofreeze)(patch listener)", true, true, true)
+runBaseTest("es5 (no freeze)", false, false)
+runBaseTest("es5 (autofreeze)", false, true)
+runBaseTest("es5 (autofreeze)(patch listener)", false, true, true)
+
+function runBaseTest(name, useProxies, autoFreeze, useListener) {
+	const listener = useListener ? function() {} : undefined
+	const {produce, produceWithPatches} = createPatchedImmer({
```

Comment:
I think to prevent bugs in future, it would be better be inside `beforeEach`. Otherwise, If `new Immer()` isn't 100% clean from global environment it will causes bugs in testing, as was before